eq-author/src/App/QuestionnaireDesignPage/NavigationSidebar/index.js

```diff
@@ -48,146 +62,160 @@ const AccordionGroupToggle = styled(Button).attrs({
-    });
+  const calculatePageErrors = pages => {
+    let count = 0;
 
-    accordionsRef.current = accordions;
+    pages.map(
+      ({ validationErrorInfo }) => (count += validationErrorInfo.totalCount)
+    );
 
-    dispatch({ type: sidebarActionTypes.handleClick });
+    return count;
   };
 
-  const handleAccordionChange = event => {
-    let accordions;
-
-    if (accordionsRef.current) {
-      accordions = accordionGroupReducer(accordionsRef.current, {
-        type: accordionActionTypes.update,
-        payload: { event },
-      });
-    } else {
-      accordions = accordionGroupReducer(questionnaire.sections, {
-        type: accordionActionTypes.createAndUpdate,
-        payload: {
-          event,
-          isOpen: true,
-        },
-      });
+  const buildPageList = ({
+    id: pageId,
+    alias,
+    title,
+    confirmation,
+    pageType,
+    validationErrorInfo,
+  }) => {
+    const components = [];
+
+    if (pageType === "QuestionPage") {
+      components.push(
+        <NavItem
+          key={title}
+          title={alias || title || "Untitled question"}
+          titleUrl={`#${buildPagePath({
+            questionnaireId: questionnaire.id,
+            pageId,
+            tab: "design",
+          })}`}
+          active={isCurrentPage(pageId, entityId)}
+          icon={IconQuestionPage}
+          errorCount={validationErrorInfo.totalCount}
+        />
+      );
     }
 
-    accordionsRef.current = accordions;
-
-    const allOpen = accordionsRef.current.every(item => item.isOpen === true);
+    if (pageType === "CalculatedSummaryPage") {
+      components.push(
+        <NavItem
+          key={title}
+          title={alias || title || "Untitled calculated summary"}
+          titleUrl={`#${buildPagePath({
+            questionnaireId: questionnaire.id,
+            pageId,
+            tab: "design",
+          })}`}
+          active={isCurrentPage(pageId, entityId)}
+          icon={IconSummaryPage}
+          errorCount={validationErrorInfo.totalCount}
+        />
+      );
+    }
 
-    if (allOpen !== state.label) {
-      dispatch({
-        type: sidebarActionTypes.toggleLabel,
-      });
+    if (confirmation) {
+      const { id: confirmationId, title: confirmationTitle } = confirmation;
+      components.push(
+        <NavItem
+          key={confirmationTitle}
+          title={confirmationTitle || "Untitled confirmation question"}
+          titleUrl={`#${buildConfirmationPath({
+            questionnaireId: questionnaire.id,
+            confirmationId,
+            tab: "design",
+          })}`}
+          active={isCurrentPage(confirmationId, entityId)}
+          icon={IconConfirmationPage}
+        />
+      );
     }
+    return components;
+  };
+
+  const buildNavigationList = ({ sections }) => {
+    const components = sections.map(
+      ({ id: sectionId, title, folders, validationErrorInfo }) => {
+        const allPagesInSection = flatMap(folders.map(({ pages }) => pages));
```

Comment:
```suggestion
        const allPagesInSection = folders.flatMap(({ pages }) => pages);
```
cut out the middle person